service/resource/deployment/deployment.go

```diff
@@ -50,12 +50,17 @@ func (r Resource) newMainDeployment(obj providerv1alpha1.AzureConfig) (deploymen
 		workerNodes = append(workerNodes, n)
 	}
 
-	masterCloudConfig, err := r.cloudConfig.NewMasterCloudConfig(obj)
+	cluster, err := r.randomkeysSearcher.SearchCluster(key.ClusterID(obj))
 	if err != nil {
 		return deployment{}, microerror.Mask(err)
 	}
 
-	workerCloudConfig, err := r.cloudConfig.NewWorkerCloudConfig(obj)
+	masterCloudConfig, err := r.cloudConfig.NewMasterCloudConfig(obj, string(cluster.APIServerEncryptionKey))
+	if err != nil {
+		return deployment{}, microerror.Mask(err)
+	}
+
+	workerCloudConfig, err := r.cloudConfig.NewWorkerCloudConfig(obj, string(cluster.APIServerEncryptionKey))
```

Comment:
The encryption key is only needed by the master. This is because its set via a flag on the api-server.

Reply:
Yes, it was my broken thinking back then and it sneaked. It would be nice to fix that in scope of that change.